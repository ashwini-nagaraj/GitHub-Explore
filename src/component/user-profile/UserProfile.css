.user__profile {
  padding-bottom: 15px;
  margin-bottom: 15px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20px;
}
.user__profileDetails {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-left: 1px;
}
.user__profile {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin-left: 1px;
}
.user__profileDetail {
  font-size: 14px;
  margin-right: 15px;
  color: #201717;
  display: flex;
  align-items: center;
  margin-left: 1px;
}

h4 {
  font-weight: bold;
}

h5 {
  color: #586069;
  font-weight: lighter;
}
