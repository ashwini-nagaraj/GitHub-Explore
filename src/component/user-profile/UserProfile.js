import React from "react";
import "./UserProfile.css";

const UserProfile = ({ user = [], starredRepositories= [] }) => (
  <div data-testid="user" className="user__profile">
    <img
      data-testid="avatar"
      src={user.avatar_url}
      className="p-10 avatar"
      width="130"
      height="130"
      alt="User"
    />
    <div data-testid="stats" className="user__profileDetails">
      <h4 data-testid="name">{user.name}</h4>
      <h5 data-testid="login">{user.login}</h5>
      <h7 data-testid="bio">{user.bio}</h7>
      <div data-testid="profile" className="user__profile">
        <div data-testid="profile_detail" className="user__profileDetail">
          <i className="fa fa-group" />
          <span>{user.followers} Followers </span>
        </div>
        <div className="user__profileDetail">
          <i className="fa fa-heart mr-5" />
          <span>{user.following} Following</span>
        </div>
      </div>
      <div>
        {user.location && (
          <div className="user__profileDetail">
            <i className="fa fa-map-marker mr-5" />
            <span>{user.location}</span>
          </div>
        )}

        {user.blog && (
          <div className="user__profileDetail">
            <i className="fa fa-globe mr-5" />
            <span>{user.blog}</span>
          </div>
        )}

        {user.email && (
          <div className="user__profileDetail">
            <i className="fa fa-envelope mr-5" />
            <span>{user.email}</span>
          </div>
        )}
        {user.twitter_username && (
          <div className="user__profileDetail">
            <i className="fa fa-twitter mr-5" />
            <span>{user.twitter_username}</span>
          </div>
        )}

        {user.orgs && (
          <div className="user__profileDetail">
            <i className="fa fa-twitter mr-5" />
            <span>{user.orgs}</span>
          </div>
        )}
        {starredRepositories.length > 0 && (
          <div className="user__profileDetail">
            <i className="fa fa-star mr-5" />
            <span>{starredRepositories.length}</span>
          </div>
        )}
      </div>
    </div>
    Â 
  </div>
);

export default UserProfile;
