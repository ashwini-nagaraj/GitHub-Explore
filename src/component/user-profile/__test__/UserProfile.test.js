import React from "react";
import ReactDOM from "react-dom";
import UserProfile from "../UserProfile";
import { render , cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashig", () => {
  const div = document.createElement("div");
  ReactDOM.render(<UserProfile></UserProfile>, div);
});

it("renders loading correctly", () => {
  const { getByTestId } = render(
    <UserProfile>
    </UserProfile>
  );
  const user = getByTestId("user");
  const avatar = getByTestId("avatar");
  expect(user).toContainElement(avatar);
});

it("matches snapshot",()=>{
    const tree=  renderer.create(<UserProfile></UserProfile>).toJSON();
     expect(tree).toMatchSnapshot();
 })