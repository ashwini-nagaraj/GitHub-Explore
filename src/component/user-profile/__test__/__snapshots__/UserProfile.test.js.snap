// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="user__profile"
  data-testid="user"
>
  <img
    alt="User"
    className="p-10 avatar"
    data-testid="avatar"
    height="130"
    width="130"
  />
  <div
    className="user__profileDetails"
    data-testid="stats"
  >
    <h4
      data-testid="name"
    />
    <h5
      data-testid="login"
    />
    <h7
      data-testid="bio"
    />
    <div
      className="user__profile"
      data-testid="profile"
    >
      <div
        className="user__profileDetail"
        data-testid="profile_detail"
      >
        <i
          className="fa fa-group"
        />
        <span>
           Followers 
        </span>
      </div>
      <div
        className="user__profileDetail"
      >
        <i
          className="fa fa-heart mr-5"
        />
        <span>
           Following
        </span>
      </div>
    </div>
    <div />
  </div>
  Â 
</div>
`;
